Feat support more link images

qoopa opened the pull request:
This adds support for images collected from link meta data

qoopa commented:
Note that these images are not loaded as fast.

gkasdorf commented:
I think it's a fair tradeoff and it shouldn't affect anything regarding FeedItem size since they will be fixed sizes. No need to worry about recycling.

andreasingman commented:
I can make another PR later to add the same thing for the default FeedItem.
And break this if appart. It's getting hard to read :P

qoopa commented:
@gkasdorf Fixed the Feed item, Post content link preview too.

gkasdorf commented:
I'm going to play with this later and see if we can't clean up the rendering a bit. I like the previews for sure, but there's definitely some stutter in the rendering. Might be able to find a solution though.

sgriff96 reviewed (changes requested):
I'm getting some errors if i scroll too fast i think
![image](https://github.com/gkasdorf/memmy/assets/46509416/ea563d7c-9a13-49d0-88c4-247cfcd83a48)

Also am still seeing a bunch of these links so I don't understand if that's trying to fix this or not?
![image](https://github.com/gkasdorf/memmy/assets/46509416/c1c7d642-4dc8-4cda-9ec9-fcef9e9affab)

gkasdorf commented:
> 

Yea this could be a solution. Especially given the long load times, we don't want to have a "flashy" UI. At the most I'd say...if possible...to load some preview text or something. Full preview can be in the post body when you open the post up.

teflocarbon commented:
I think it’s fine showing the description and the title along with the image, as long as it’s only a small amount just to give the reader an idea. These come from the html meta tags, specially Open Graph which is by Meta. It’s commonly used on news articles so that where they’re posted to platforms like Facebook and Twitter they have rich metadata, some don’t use it though.

I’d probably want the view to be cleaner though, having square corners when most things are rounded looks off.

https://ogp.me/

qoopa commented:
Some site does not return the data soo that’s when it fallbacks to the `linkbutton` 

Also agree the stutter in the feed becomes a bit annoying, originally started with compact feed where the problem doesn’t really exist. 

Even we find a way around the stutter I think this could be behind a setting also.

qoopa commented:
Made a minimal link preview that is the same size as old and fixed, so should not flicker.

Then a more full version for post.

Also added a settings so it would have to be turned on to be used.

<table>
<tr><td>Feed preview link</td><td>Post preview link</td></tr>


<tr><td>
<img src="https://github.com/gkasdorf/memmy/assets/2129171/8e47c299-1276-411a-8798-6326887c8aee" width="300" />
</td><td>
<img src="https://github.com/gkasdorf/memmy/assets/2129171/f2121b52-a8b1-445b-a245-fa458f7b61d0" width="300" />
</td></tr>

</table>

gkasdorf commented:
Oh this looks a lot better. I'll take a look at it later today and merge it in if everything is good. Thanks!

sgriff96 commented:
@gkasdorf and I talked, we want to use the new `LinkPreviewContainer` for the `PostContentView` exclusively, and then keep only using `LinkButton` for `FeedContentPreview`. I think I'm having some weird render and performance issues with the compact feed though. And I don't really understand what changed with the compact feed, I am seeing basically the same stuff in the simulator as I am on the current TestFlight version.

gkasdorf commented:
> 

This *could* have been me. I'll message you on Discord.

gkasdorf commented:
This might be a simulator issue with performance/rendering. We probably should test to be sure on an actual build.

sgriff96 commented:
Ah yea having the same issue on main so nvm about those perf issues.

qoopa commented:
Closing this since interest seems low
